D3D12HelloWorld/Graphics/Rendering.cs

```diff
@@ -16,19 +16,27 @@ public abstract class ResourceView {
     }
 
     public class ConstantBufferView : ResourceView {
-        public ConstantBufferView(GraphicsResource resource) : base(resource, CreateConstantBufferView(resource)) {
+        public ConstantBufferView(ConstantBufferView constantBufferView) : base(constantBufferView.Resource, constantBufferView.CpuDescriptorHandle) {
         }
 
-        public ConstantBufferView(ConstantBufferView constantBufferView) : base(constantBufferView.Resource, constantBufferView.CpuDescriptorHandle) {
+        public ConstantBufferView(GraphicsResource resource)
+            : this(resource, 0, resource.SizeInBytes, resource.GraphicsDevice.ShaderResourceViewAllocator) {
         }
 
-        private static CpuDescriptorHandle CreateConstantBufferView(GraphicsResource resource) {
-            CpuDescriptorHandle cpuHandle = resource.GraphicsDevice.ShaderResourceViewAllocator.Allocate(1);
+        internal ConstantBufferView(GraphicsResource resource, ulong offset, ulong sizeInBytes, DescriptorAllocator shaderResourceViewAllocator)
+            : base(resource, CreateConstantBufferView(resource, offset, sizeInBytes, shaderResourceViewAllocator)) {
+        }
+
+        public static ConstantBufferView FromOffset<TBuffer>(GraphicsResource resource, ulong offset, DescriptorAllocator shaderResourceViewAllocator)
+            => new ConstantBufferView(resource, offset, (ulong)Unsafe.SizeOf<TBuffer>(), shaderResourceViewAllocator);
```

Comment:
This new internal constructor, along with a public static method that wraps it, now allows creation of CBVs using an alternate DescriptorAllocator (such as a shader-visible one, for resources already uploaded to the GPU, as is often found in the C++ DirectX samples from Microsoft)